platforms/core-configuration/configuration-cache/src/integTest/groovy/org/gradle/internal/cc/impl/isolated/CompositeBuildSupport.groovy

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.cc.impl.isolated
+
+import org.gradle.test.fixtures.file.TestFile
+
+import java.util.function.Function
+
+trait CompositeBuildSupport {
```

Comment:
🤔  The convention in the Gradle codebase seems to be calling such classes `Fixture`

Reply:
The purpose of this class is to provide a reusable piece of logic used in tests. This is a "fixture" in Gradle's terminology, and that's why we have `testFixtures` source sets.

The fact that this fixture is a trait is an implementation detail.